pkg/proxy/proxier.go

```diff
@@ -463,28 +464,29 @@ func (proxier *Proxier) OnUpdate(services []api.Service) {
 	glog.V(4).Infof("Received update notice: %+v", services)
 	activeServices := util.StringSet{}
 	for _, service := range services {
-		activeServices.Insert(service.Name)
-		info, exists := proxier.getServiceInfo(service.Name)
+		cachekey := types.NewNamespacedName(service.Namespace, service.Name).CacheKey()
```

Comment:
You'r right that "New" typically means pointer, but not always.  I prefer to think of it as returning a useful and safe-to-copy value.  Given the raison d'etre of this type, the "reasonable value" is a value, not a pointer.

C++ has a convention of "New" meaning pointer and "Make" meaning value.  I would be ok with that, but I don't think it is necessary.

I think that using the object as key is A+, and I think CacheKey() serves no purpose.  Just return a value.  But also read my response to the "public fields" point.